augur/utils.py

```diff
@@ -675,19 +691,56 @@ def read_bed_file(bed_file):
     list of int:
         Sorted list of unique zero-indexed sites
     """
-    mask_sites = []
-    try:
-        bed = pd.read_csv(bed_file, sep='\t', header=None, usecols=[1,2],
-                          dtype={1:int,2:int}, **PANDAS_READ_CSV_OPTIONS)
-    except ValueError:
-        # Check if we have a header row. Otherwise, just fail.
-        bed = pd.read_csv(bed_file, sep='\t', header=None, usecols=[1,2],
-                          dtype={1:int,2:int}, skiprows=1, **PANDAS_READ_CSV_OPTIONS)
-        print("Skipped row 1 of %s, assuming it is a header." % bed_file)
-    for _, row in bed.iterrows():
-        mask_sites.extend(range(row[1], row[2]))
+    in_header = True
+    initial_chrom_value = ""
```

Comment:
It seems valid to have an empty string in `chrom` so I wouldn't use that as a placeholder value (currently the single unique value validation does not error if the first entry has `chrom=""` followed by `chrom="<anything else>"`).

Suggestion: set `initial_chrom_value` to `None`

```suggestion
    initial_chrom_value = None
```

and check that explicitly further down:

```diff
-             if initial_chrom_value:
+             if initial_chrom_value is not None:
                if chrom != initial_chrom_value:
                    raise AugurError(
```